paf-mvp-frontend/src/services/http.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Log } from '@core/log';
+
+export interface IHttpService {
+  /** Yes if the service has started a redirection. */
+  isRedirecting: boolean;
+  /**
+   * Start a redirection on the given URL if the service
+   * hasn't started already one.
+   */
+  redirect(url: string): boolean;
+  /**
+   * Stringify the JS object and HTTP-POST it at the given URL
+   * with headers including 'credentials: include'.
+   *
+   * For now, we don't use Content-type JSON to avoid CORS pre-flight request.
+   * See https://stackoverflow.com/questions/37668282/unable-to-fetch-post-without-no-cors-in-header
+   */
+  postJson(url: string, input: object): Promise<Response>;
+  /**
+   * Call HTTP POST the given body at the given URL
+   * with headers including 'credentials: include'.
+   */
+  postText(url: string, input: string): Promise<Response>;
+  /**
+   * Call HTTP GET at the given URL
+   * with headers including 'credentials: include'.
+   */
+  get(url: string): Promise<Response>;
+  /**
+   * Call HTTP DELETE at the given URL
+   * with headers including 'credentials: include'.
+   */
+  deleteHttp(url: string): Promise<Response>;
+}
+
+type Fetcher = typeof fetch;
+
+export class HttpService implements IHttpService {
+  fetch: Fetcher;
+  logger: Log;
+  isRedirecting: boolean;
+
+  constructor(fetcher: Fetcher = fetch, logger = new Log('OneKey', '#3bb8c3')) {
```

Comment:
INT FAR. Favor more compact declaration:

```typescript
  isRedirecting: boolean = false;

  constructor(public fetcher: Fetcher = fetch, public logger = new Log('OneKey', '#3bb8c3')) {}
```

Reply:
I prefer the current version and I suggest the following rule: if there is at least one property that is not injected in the constructor, then we should avoid using property declarations in the parameters of the constructor.

The goal is to have a clear vision by gathering properties in the same place.